"use strict";

/**
 * This function takes an async express route / middleware as a parameter and wraps it in a promise.
 * This allows to implicitly catch all errors thrown in the wrapped async route / middleware
 * and pass them forward to the ErrorHandler.
 * @param fn
 */
module.exports = fn => (req, res, next) => {
  Promise.resolve(fn(req, res, next)).catch(next);
};
